src/d3d9/d3d9_swapchain.cpp

```diff
@@ -727,8 +737,23 @@ namespace dxvk {
     m_parent->Flush();
 
     // Retrieve the image and image view to present
-    auto swapImage = m_backBuffers[0]->GetCommonTexture()->GetImage();
-    auto swapImageView = m_backBuffers[0]->GetImageView(false);
+    Rc<DxvkImage> swapImage;
+    Rc<DxvkImageView> swapImageView;
+
+    const bool copyToFrontbuffer = m_presentParams.SwapEffect == D3DSWAPEFFECT_COPY
+      || m_presentParams.SwapEffect == D3DSWAPEFFECT_COPY_VSYNC
+      || m_backBuffers.size() == 2;
```

Comment:
This m_backbuffers.size() == 2 thing seems sus. I guess its really because there is only one back buffer + one frontbuffer?

Reply:
Yeah that's the idea.

Its something I noticed when looking at an apitrace of one of the games we enable noExplicitFrontbuffer for.

Reply:
Wonder how many games inadvertently hit this though... Hmmm.

Reply:
What swap effect was the game using?

Reply:
That's the fun part. SWAPEFFECT_DISCARD...

Reply:
Maybe our handling of frontbuffer is wrong and there really is only meant to be one backbuffer which is also the frontbuffer? Maybe only COPY has an implicit extra frontbuffer... idk.